common/src/main/scala/uk/ac/wellcome/platform/stacks/common/services/CatalogueService.scala

```diff
@@ -1,158 +1,137 @@
 
-  protected def getItemIdentifiersFrom(item: ResultListItems): List[ItemIdentifiers] =
-    Option(item.getIdentifiers)
-      .map(_.asScala.toList)
-      .getOrElse(Nil)
-
-  protected def getStacksItemIdentifierFrom(item: ResultListItems): Future[Option[StacksItemIdentifier]] = {
-    val identifier = getItemIdentifiersFrom(item)
-      .filter { _.getIdentifierType.getId == "sierra-identifier" }
-      .map { _.getValue }
-
-    identifier match {
-      case List(sierraItemId) => {
-        val catalogueId = CatalogueItemIdentifier(item.getId)
-        val sierraId = SierraItemIdentifier
-          .createFromString(sierraItemId)
-
-        Future.successful(
-          Some(StacksItemIdentifier(catalogueId, sierraId))
-        )
-      }
-      case Nil => Future.successful(None)
-      case _ => Future.failed(new Throwable(
-        f"Ambiguous identifier! ($identifier)"
-      ))
-    }
+  protected def getLocations(
+                              locations: List[LocationStub]
+                            ): List[StacksLocation] = locations collect {
+    case location@LocationStub(_, _, "PhysicalLocation") =>
+      StacksLocation(
+        location.locationType.id,
+        location.locationType.label
+      )
   }
 
-  protected def getStacksLocationFrom(item: ResultListItems): Future[Option[StacksLocation]] = {
-    val itemLocations = item.getLocations.asScala.toList
-
-    val physicalLocation = itemLocations
-      .map(_.asInstanceOf[LTM[Any]])
-      .map { l =>
-        (
-          l.safeGet[LTM[String]]("locationType").map(_.get("id")),
-          l.safeGet[String]("type"),
-          l.safeGet[String]("label")
-        )
-      }
-      .collect {
-        case (Some(id), Some("PhysicalLocation"), Some(label)) =>
-          StacksLocation(id, label)
-      }
-
-    physicalLocation match {
-      case List(location) => Future.successful(Some(location))
-      case Nil => Future.successful(None)
-      case _ => Future.failed(new Throwable(
-        f"Ambiguous location! ($itemLocations)"
-      ))
-    }
+  protected def getStacksItems(
+                                itemStubs: List[ItemStub]
+                              ): List[StacksItemWithOutStatus] = itemStubs map {
+    case ItemStub(id, identifiers, locations) =>
+      (
+        CatalogueItemIdentifier(id),
+        getIdentifier(identifiers),
+        getLocations(locations)
+      )
+  } map {
+    case (catId, Some(sierraId), List(location)) =>
+      StacksItemWithOutStatus(
+        StacksItemIdentifier(catId, sierraId),
+        location
+      )
   }
 
-  private def getItems(identifier: Identifier[_]): Future[List[StacksItemWithOutStatus]] = {
+  def getStacksWork(workId: StacksWorkIdentifier): Future[StacksWork[StacksItemWithOutStatus]] = {
+
+    val urlPath = s"works/${workId.value}"
+    val queryParams = "include=items%2Cidentifiers"
+    val baseUri = maybeBaseUri.getOrElse(defaultBaseUri)
+    val uri = s"${baseUri}/${urlPath}?${queryParams}"
```

Comment:
The akka-http DSL has some pieces that might make this a little cleaner:

```scala
val query = Query(
  ("include", "items,identifiers")
)
    
val baseUri = maybeBaseUri.getOrElse(defaultBaseUri)
    
val uri = baseUri
  .copy(path = baseUri.path + s"works/${workId.value}")
  .withQuery(query)
```

Reply:
The downstream PR moves this code around a bit - I agree these changes are sensible, but i'm tempted to make them downstream so as to avoid merge conflicts.

Reply:
:nod:

Makes sense. The code is fine as-is, but using the Query DSL is a bit cleaner and would be nice to do if/when it's convenient.